libethcore/Precompiled.cpp

```diff
@@ -132,15 +132,34 @@ ETH_REGISTER_PRECOMPILED(modexp)(bytesConstRef _in)
 	return {true, ret};
 }
 
+namespace
+{
+	bigint expLengthAdjust(bigint _baseLength, bigint _expLength, bytesConstRef _in)
+	{
+		size_t const numConsideredBytes(min(_expLength, {32}));
```

Comment:
I'd say this `numConsideredBytes` trick and having one return statement makes it pretty hard to read.

I'd express this somewhat closer to EIP spec with something like this:
```
if (_expLength <= 32)
{
	bigint const exp(parseBigEndianRightPadded(_in, 96 + _baseLength, expLength));
	return exp ? msb(exp) : 0;
}
else
{
	bigint const expFirstWord(parseBigEndianRightPadded(_in, 96 + _baseLength, 32));
	size_t const highestBit = expFirstWord ? msb(expFirstWord) : 0;
	return 8 * (_expLength - 32) + highestBit;
}
```